### ISSUE TITLE
Reply if you wanna contribute

### ISSUE THREAD

**Issue by PeerRich** [NostrGit/NostrGit#145]
### Discussed in https://github.com/NostrGit/NostrGit/discussions/77

<div type='discussions-op-text'>

<sup>Originally posted by **PeerRich** March  5, 2023</sup>
Jack Dorsey made a bounty

https://news.ycombinator.com/item?id=35020964

We'd love to split it with everyone contributing.

Comment below and go through the [Issues](https://github.com/NostrGit/NostrGit/issues) to see where you can help</div>

**Comment by melvincarvalho** [comment]
:+1: see: https://github.com/NostrGit/NostrGit/discussions/77#discussioncomment-5227558

**Comment by RandyMcMillan** [comment]
I have been thinking about this - even before @jack posted the bounty...
Plenty of ideas...
Hope to contribute where I can... :) 
https://twitter.com/search?q=%40randymcMillan%20nostr%20git&src=typed_query

**Comment by cypherhoodlum** [comment]
Awesome @RandyMcMillan, looking forward to your contributions!

**Comment by pierresisson** [comment]
welcome everybody

**Comment by cypherhoodlum** [comment]
Hi, @alltheseas. @PeerRich  is the one who can give you the necessacy access. My guess is he will get back to you during the weekend. In the meantime it would be of great help to try to conceptualize features as you have done in the mentioned user story. Right now the biggest issue is to reach consensus about the structure of nostrgit repositories, how to create them, how to manage their state, and where the code should be hosted. A smaller but very important thing is to figure out how Issues work on nostrgit. They should probably be nostr text_events referencing a nostrgit repository. Then there's the pull request structure to figure out. And the zappable interactions of all kinds. These could all be made into structured milestones and/or documented conceptually.

**Comment by PeerRich** [comment]
@alltheseas welcome! added you with write access

**Comment by pqingshuang** [comment]
Cool! Looking forward about how this project can work together with nostrocket@!!!!

**Comment by arikchakma** [comment]
I wanna join this awesome journey.

**Comment by cypherhoodlum** [comment]
Welcome @arikchakma :) you can find all the necessary info in the readme. Give a look at the contribution guidelines linked there to get started. Looking forward to your contributions :rocket:

**Comment by arjunbharti** [comment]
Hey yo, would love to contribute!

**Comment by kalmndo** [comment]
i'm so excited to contribute this project :rocket:

**Comment by cypherhoodlum** [comment]
Looking forward to your contributions @arjunbharti @kalmndo :rocket:

**Comment by Rajashekarksreddy** [comment]
I am very much excited to contribute this project!

**Comment by Just-Moh-it** [comment]
Hey @PeerRich would love to contribute! Mostly a t3-stack full-stack guy, so would be pretty useful!

**Comment by eesazahed** [comment]
I'd like to help contribute with some front-end stuff

**Comment by eesazahed** [comment]
I'm interesting in contributing regarding some UI issues

**Comment by Inalegwu** [comment]
I would love to

**Comment by tobiadey** [comment]
👀

**Comment by AntwanSherif** [comment]
Very excited to contribute

**Comment by akhilesh-w** [comment]
Hey! How can I help

**Comment by cypherhoodlum** [comment]
Hi, @akhilesh-w.
You can start by reading the readme and the contribution guidelines linked there to get started.
After that you can pick some unassigned Issue from the [issues tab](https://github.com/NostrGit/NostrGit/issues) and start working on that. Alternatively, when you find a functionality that's clearly missing from nostrgit.com, [create a new issue](https://github.com/NostrGit/NostrGit/issues/new) for it and start working on that.

**Comment by Master-Tech271** [comment]
I'm very excited to contribute on this project and utilize my skills

**Comment by hazeycode** [comment]
Hey, I'd like to contribute to this most excellent endeavor

**Comment by LvckyAPI** [comment]
i realy want to help

**Comment by ishanavasthi** [comment]
Hi I am in

**Comment by itstomekk** [comment]
Guys don't give up! I hope someone has time & resources to take up on this :)

**Comment by cypherhoodlum** [comment]
It's good to have encouragement, thanks @itstomekk :rocket: The development on this repo has slowed down (in my opinion) due to lack of some crucial fundamentals, namely the file storage solution for repos on nostr relays. That problem is being resolved [here](https://github.com/HORNET-Storage). You can read more about scionic merkle DAGs (the solution to the file storage problem) [here](https://www.hornetstorage.com/). I will be focusing on building NostrGit with the HORNET-storage team, so NostrGit will still happen! We might need some help with the grunt work because the team is small. Cheers!

edit (Mar 19, 2025):
The project called Nestr nowadays and we're very close to delivering a full repo management solution on Nostr.

**Comment by cypherhoodlum** [comment]
If you're serious about contributing to the project mentioned above (probably frontend & nostr stuff), you can reach out to Colby Serpa (npub1t89vhkp66hz54kga4n635jwqdc977uc2crnuyddx7maznwfrpupqwra5h9) or me (npub1u9e887ad8pl49cxgzqkuljxcxy89dtac7jkyuajnukxg6hu2hufqdjdsp7).

**Comment by Austinati** [comment]
Maybe I would have some time to work on this?

**Comment by melvincarvalho** [comment]
> Maybe I would have some time to work on this?

Awesome!  The project is slightly dormant right now, and could do with someone to help drive it forward.

**Comment by melvincarvalho** [comment]
yes see: https://github.com/NostrGit/NostrGit/issues/145#issuecomment-1562528119
